### PR TITLE
Path resolving fixes

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

The recent addition of path validation made resolving broken. This change moves up the path resolution closer to the user imput.

## Steps to Test

Try different paths when creating new environment for local components.

### CODE REVIEW

File: src/lib/dev-environment/dev-environment-cli.js [hunk 1]
```diff
@@ -183,6 +194,13 @@ async function processComponent( component: string, option: string ) {
 	return result;
 }
 
+export function resolvePath( input: string ): string {
+	// Resolve does not do ~ reliably
+	const resolvedPath = input.replace( /^~/, os.homedir() );
```
Comment (pauarge) [inline src/lib/dev-environment/dev-environment-cli.js:199]: Is this already handled by some built-in function?

File: src/lib/dev-environment/dev-environment-cli.js [hunk 2]
```diff
@@ -273,7 +291,7 @@ export async function promptForComponent( component: string ) {
 	};
 }
 
-async function getLatestImageTags( component: string ): string[] {
+function getLatestImageTags( component: string ): string[] {
```
Comment (pauarge) [inline src/lib/dev-environment/dev-environment-cli.js:294]: 👍

File: src/lib/dev-environment/dev-environment-cli.js (hunk 1 above)
Comment (pschoffer) [reply src/lib/dev-environment/dev-environment-cli.js:199]: Doesn't seem to from what I have found online - https://stackoverflow.com/a/37112508/11875271

Like there are some tools probably, but it seems kinda simple to introduce a new dependency.

**Review by pauarge — APPROVED** [review]
LGTM

**Pull request merged**